### ISSUE TITLE
Add Trino to ODH

### ISSUE THREAD

**Issue by rimolive** [operate-first/support#119]
**Is your feature request related to a problem? Please describe.**
Currently, Data Catalog can't move to some features like Access Control, Data Governance because of the Thrift Server limitations. We are considering a change to Trino as it offers all the features we need.

**Describe the solution you'd like**
Let's start with adding Presto in the  ODH manifests and work on integration with the Data Catalog components (Hue, Superset)

**Describe alternatives you've considered**
Thrift Server was the alternative but after evaluation, we made the conclusion that it is very limited according to our needs.

**Additional context**
N/A

**Comment by accorvin** [comment]
cc @durandom

**Comment by accorvin** [comment]
We should be able to reuse the work that the cost management and ansible tower teams have done to install Trino on OCP for this. Theoretically we could Kustomize their deployment manifests and add them into the ODH.

**Comment by tumido** [comment]
If you want to use OperateFirst as a test bed for an alternative to the current ODH Datacatalog, we welcome that! :+1: How can we help?

**Comment by rimolive** [comment]
@tumido I'm currently working on the kustomize descriptors. After that, I would need help with start to write some basic referential architecture docs.

**Comment by accorvin** [comment]
@rimolive presumably we'd also like some help testing this in op-first?

**Comment by rimolive** [comment]
Yeah, this would be good too.

**Comment by accorvin** [comment]
FYI we are working on the pull request for this here: https://github.com/opendatahub-io/odh-manifests/pull/373

**Comment by tumido** [comment]
@accorvin @rimolive if you want it deployed even before it's merged into ODH, we can create a namespace for it and deploy a kfdef against your PR branch. WDYT?

**Comment by accorvin** [comment]
That would be great @tumido 

I'm actually blocked right now on not having access to my development environment to test my changes (due to the PSi outage). Can I use the MOC for this?

**Comment by tumido** [comment]
@accorvin sure, let me create a namespace for you and add you to the operate-frist user group so you can manage that namespace. https://github.com/operate-first/apps/pull/479

Let me know if you can create a kfdef for Trino in that namespace once the PR is merged, I'm worried that will be blocked (if I remember correctly) since we recently limited users from creating kfdefs due to security reasons. We may want to create an extra cluster role and bind it to these "stage-like" namespaces so certain users can change `kfdef`s... Why I'm thinking out loud instead of writing that cluster role... :smile:

**Comment by accorvin** [comment]
Thanks @tumido! For now I'll just be manually applying the kustomize objects. Once I have Trino functional we can test it via the kfdef.

**Comment by tumido** [comment]
@accorvin the namespace `opf-trino` is available for you on the zero cluster now.

**Comment by durandom** [comment]
@rimolive @tumido did we end up installing ODH trino?

https://github.com/opendatahub-io/odh-manifests/tree/master/trino

**Comment by rimolive** [comment]
@durandom I think it was unclear who's going to deploy Trino on the operate-first cluster, but I'll work with @tumido to have Trino deployed soon.

**Comment by tumido** [comment]
@durandom @rimolive nope, I did not deploy Trino. My comments and involvement here was to enable @accorvin to be able to install it.

@rimolive if you can prepare a minimal kfdef manifest just for Trino we can deploy it from ODH pretty, would you like to submit a PR for that?

It should go into here:
https://github.com/operate-first/apps/tree/master/kfdefs

I'll do a PR granting you access to that namespace so you can debug things on the way, wdyt?

**Comment by durandom** [comment]
@hemajv wants to prototype some grafana or superset dashboards that connect to parquet files from the black flake project

**Comment by HumairAK** [comment]
> It should go into here:
Relevant: 
https://github.com/operate-first/apps/blob/master/docs/odh/adding_kfdefs.md

**Comment by hemajv** [comment]
> @hemajv wants to prototype some grafana or superset dashboards that connect to parquet files from the black flake project

yes! Would be more than happy to test it out by connecting to superset/grafana for creating dashboards as currently we are unable to have visualizations in superset

**Comment by goern** [comment]
@pacospace FYI

**Comment by rimolive** [comment]
I suppose this issue can be closed since we have already Trino deployed. WDYT?

**Comment by accorvin** [comment]
I agree. Will let someone from op first chime in.

**Comment by 4n4nd** [comment]
@rimolive @accorvin We are waiting on this PR upstream: https://github.com/opendatahub-io/odh-manifests/pull/454

We do have it deployed currently, but I had to change the image being used manually. once the above PR ismerged, we can update our superset kfdef to point to the master branch and directly deploy it from upstream.

**Comment by HumairAK** [comment]
Looks like we're good to close this out, special thanks to @rimolive for all his help and continued guidance.
